independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.multistage.native

```diff
@@ -0,0 +1,36 @@
+####
+# This Multistage Dockerfile is used in order to build a container that runs the Quarkus application in native (no JVM) mode.
+#
+# Build the image with:
+#
+# docker build -f src/main/docker/Dockerfile.multistage.native -t quarkus/{project.artifact-id} .
+#
+# Then run the container using:
+#
+# docker run -i --rm -p 8080:8080 quarkus/{project.artifact-id}
+#
+###
+
+## Stage 1 : build with Maven builder image
+FROM {dockerfile.builder.from} AS build
+ARG MAVEN_BUILD_EXTRA_ARGS=
+COPY --chown=quarkus:quarkus src mvnw pom.xml /code/
+COPY --chown=quarkus:quarkus .mvn/ /code/.mvn/
+USER quarkus
+WORKDIR /code
+RUN ./mvnw -B -ntp org.apache.maven.plugins:maven-dependency-plugin:3.5.0:go-offline
```

Comment:
Does this go-offline code actually do anything in this case? To make this pattern work I thought you need to copy the pom.xml, then run go-offline, then copy the source. This way if there is no pom.xml change docker can re-use the layer. If you have copied everything then every change will re-download. 

Also you would likely want quarkus:go-offline to actually get all the deployment side dependencies.


In this case though you are copying everything, so any code change will likely result in a full download again.

Reply:
Ah, good catch. `quarkus:go-offline` is better too, I'll change to that if we all agree this PR is a good idea